File: day01/part1.py
def compute(data):
    """
    >>> compute(("1721", "979", "366", "299", "675", "1456"))
    514579
    """
    nums = {*map(int, data)}
    for num1 in nums:
        if (num2 := 2020 - num1) in nums:
            break
    else:
        assert False

    return num1 * num2


def main():
    import pathlib

    input_path = pathlib.Path(__file__).with_name("input.txt")

    with input_path.open() as f:
        print(compute(f))


if __name__ == "__main__":
    main()

File: day01/part1.exs
defmodule Day01.Part1 do
  def compute(data) do
    nums =
      data
      |> String.trim()
      |> String.split()
      |> Enum.map(&String.to_integer/1)

    nums
    |> Enum.map(&[&1, 2020 - &1])
    |> Enum.find(fn [_, b] -> b in nums end)
    |> Enum.reduce(&(&1 * &2))
  end

  def main do
    File.read!("input.txt")
    |> compute
    |> IO.inspect()
  end
end

Day01.Part1.main()
